### ISSUE TITLE
Fix `declaration-block-no-redundant-longhand-properties` false positives for properties with `!important`

### ISSUE THREAD

**Issue by dangreen** [stylelint/stylelint#8323]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	top: 0;
	right: 0;
	bottom: 0 !important;
	left: 0;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "declaration-block-no-redundant-longhand-properties": true
  }
}
```

### How did you run Stylelint?

https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLjiADvCgAwDcWuATgJYDmAFjseZTgEYR4QC2LKAQho8CEKjiSYcFbDgA2MAGbNSMgL5YQAGhCKaCgHJIecRDAAexggoB0YAM73t4CJj10EIDJhQp0ICxwYTCh7f2IAbSxfX397HABPBTkaKQBaSDd6NPjJKCQqKH9olABdLDVnTPcAMTEeJBxPACt7V2dYAidEb1iQeKSYFKlwvxA5Rph4-y0SuMTk1JwM13cciRCCooQxiaDpkAqQNSA

### Which Stylelint-related dependencies are you using?

latest

### What did you expect to happen?

```css
a {
	top: 0;
	right: 0;
	bottom: 0 !important;
	left: 0;
}
```

should not be formatted as 

```css
a {
	inset: 0;
}
```

because of `!important`

### What actually happened?

```css
a {
	top: 0;
	right: 0;
	bottom: 0 !important;
	left: 0;
}
```

formatted as

```css
a {
	inset: 0;
}
```

**Comment by Mouvedia** [comment]
I've labeled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

---

@jeddy3 I have set the type to Bug but it could also be considered an Enhancement.

**Comment by jeddy3** [comment]
@dangreen Thanks for the report and for using the template.

---

> @jeddy3 I have set the type to Bug but it could also be considered an Enhancement.

@Mouvedia Thanks for triaging the issue. The type of issue typically becomes clearer when we rename the title to match [our conventions](https://stylelint.io/maintainer-guide/issues#titles) as part of the triage. In this case, it seems like this is a false positive bug.
